addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/gui/ParamMinerDialog.java

```diff
@@ -79,14 +83,30 @@ public void init(HttpMessage target) {
             config = new ParamMinerConfig();
         }
         this.removeAllFields();
-        this.addUriSelectField(URL, this.target, true);
+        this.addUriSelectField(URL, this.target);
         this.addComboField(CONTEXT, new String[] {}, "");
         this.addCheckBoxField(FCBZ_CACHE_BUSTER, false);
         this.addCheckBoxField(PREDEFINED + URL_WORDLIST, true);
         this.addCheckBoxField(PREDEFINED + HEADER_WORDLIST, false);
         this.addCheckBoxField(PREDEFINED + COOKIE_WORDLIST, false);
 
         this.addCheckBoxField(CUSTOM + URL_WORDLIST, false);
+        this.addWordlistSelectField(CUSTOM + "url_wordlist_location", "");
+        this.addFieldListener(
+                CUSTOM + URL_WORDLIST,
+                (e) -> {
+                    if (hasClicked) {
+                        hasClicked = false;
+                        // This should remove field
+                        wordlistPanel.setVisible(false);
+                    } else {
+                        hasClicked = true;
+                        // This should add field
```

Comment:
These comments can be removed I think, the `setVisible` call should be obvious enough.